core/src/main/java/smile/classification/RBFNetwork.java

```diff
@@ -156,7 +156,7 @@ public Trainer(Metric<T> distance) {
          * @param rbf the radial basis function.
          * @param m the number of basis functions.
          */
-        public Trainer setRBF(RadialBasisFunction rbf, int m) {
+        public Trainer<T> setRBF(RadialBasisFunction rbf, int m) {
```

Comment:
One further notice, in eclipse many of these warnings are generated directly when you write the code due to the incremental compiler, I am not sure how IntelliJ IDEA is working there but I have in mind that it doesn't compile instantly when you write code but only on certain actions. What I am quite sure of, is that IntelliJ does not by default run the so callled "code inspection" all the time, you have to trigger it manually, and it has a vast amount of configuration options. They are quite powerful and should definitely find and report such things.